src/interface/sparse_pack_base.cpp

```diff
@@ -0,0 +1,265 @@
+//========================================================================================
+// (C) (or copyright) 2020-2022. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include <algorithm>
+#include <functional>
+#include <limits>
+#include <map>
+#include <memory>
+#include <regex>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "coordinates/coordinates.hpp"
+#include "interface/mesh_data.hpp"
+#include "interface/meshblock_data.hpp"
+#include "interface/sparse_pack_base.hpp"
+#include "interface/variable.hpp"
+#include "utils/utils.hpp"
+
+namespace {
+// SFINAE for block iteration so that sparse packs can work for MeshBlockData and MeshData
+template <class T, class F>
+inline auto IterateBlocks(T *pmd, F func) -> decltype(T().GetBlockData(0), void()) {
```

Comment:
This function applies `F func` to every block in the MeshBlock/MeshData. Could this be renamed to `ForEachBlock` since this functions like `std::for_each`? Could you also add a comment about what this is doing?

```suggestion
//Apply a function to each block in a MeshBlockData or MeshData
template <class T, class F>
inline auto IterateBlocks(T *pmd, F func) -> decltype(T().GetBlockData(0), void()) {
```

Reply:
Yeah, I think renaming it `ForEachBlock` is a good idea for consistency with the stdlib and for consistency with the boundary iteration machinery.

Reply:
done